.github/actions/prepare-tox/action.yml

```diff
@@ -0,0 +1,15 @@
+# A composite action to prepare tox-based workflows
+
+name: "Prepare tox"
+
+description: "Prepare tox-based workflows"
+
+runs:
+  using: "composite"
+  steps:
```

Comment:
It seems reasonable to also require installing Python here. Are there compelling reasons why choosing a standard version (3.11) and using that for all tox jobs wouldn't work?

Reply:
The reason I had not included that one was because I hadn't put thought into how the matrixing versions (in lint) and full inclusion of all versions (in the tests) should be reconciled.

When we run `tox -e py39,py310,...` it runs an individual tox job for tests for each. That means that the runner has to have all those versions of Python installed to start with, because tox will look for each Python as it runs each job. But for linting, our `tox -e lint` job will just use whichever Python is the default in our virtual environment (I think). 

The `setup-python` action seems to allow you to [specify the default Python version](https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#specifying-multiple-pythonpypy-versions)as the last version identified among multiple selected. I'd think if nothing else we might want to control which version is used for linting (as either the most recent Python version, or the second most recent if we think there's more stability benefits for that).

Also, another concern that popped into my head might be that there could be a performance penalty for installing all versions of Python for the linting job. From what I can see on the pipelines right now, it's [less than 1 second for just one version at a time](https://github.com/sandialabs/firewheel/actions/runs/11865058215/job/33069575968) and [closer to 9 seconds to install all 6 (Python 3.8 through 3.13)](https://github.com/sandialabs/firewheel/actions/runs/11865058209). That's not huge for a CI pipeline—especially if it's running concurrent with tests which will almost always take far longer—so I'm inclined to think it's not a big deal.